### PR TITLE
Accounts V2: Derived Keymanager List Accounts

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

This PR implements the `list` functionality for derived keymanager accounts

**Which issues(s) does this PR fix?**

Part of #6220

### CODE REVIEW

File: shared/petnames/names.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package petnames
```
Comment (rauljordan) [inline shared/petnames/names.go:1]: The current golang petnames library does not allow for an easy, deterministic selection of petnames. This adds the functionality as a small shared package.

File: validator/accounts/v2/wallet.go [hunk 2]
```diff
@@ -246,6 +250,20 @@ func (w *Wallet) WriteFileAtPath(ctx context.Context, filePath string, fileName
 	return nil
 }
 
+// ReadFileAtPath within the wallet directory given the desired path and filename.
+func (w *Wallet) ReadFileAtPath(ctx context.Context, filePath string, fileName string) ([]byte, error) {
```
Comment (rauljordan) [inline validator/accounts/v2/wallet.go:254]: There are too many ways to read and writes files in the wallet. In a separate PR, all of these will be simplified and deduplicated.

File: validator/keymanager/v2/derived/derived.go [hunk 3]
```diff
@@ -288,3 +385,52 @@ func (dr *Keymanager) generateKeystoreFile(privateKey []byte, publicKey []byte,
 	}
 	return json.MarshalIndent(keystoreFile, "", "\t")
 }
+
+func generateDepositTransaction(
```
Comment (rauljordan) [inline validator/keymanager/v2/derived/derived.go:389]: In a follow-up PR, ill abstract this and perform a full cleanup of the accounts v2 code for deduplication

File: validator/accounts/v2/list.go [hunk 4]
```diff
@@ -117,3 +131,82 @@ func listDirectKeymanagerAccounts(
 	fmt.Println("")
 	return nil
 }
+
+func listDerivedKeymanagerAccounts(
+	showDepositData bool,
+	wallet *Wallet,
+	keymanager *derived.Keymanager,
+) error {
+	au := aurora.NewAurora(true)
```
Comment (0xKiwi) [inline validator/accounts/v2/list.go:140]: Part of our cleanup should include making a global aurora instance, wdyt? Maybe making a `colors.go` file and doing `AccountsColor()`, `DirColor()` to make sure colors are consistently used.

**Pull request merged**